src/Transport/Addressing/Partitioning/Strategies/FailOverNamespacePartitioning.cs

```diff
@@ -3,21 +3,17 @@
     using System.Collections.Generic;
     using System.Configuration;
     using System.Linq;
+    using AzureServiceBus.Addressing.Partitioning;
     using Transport.AzureServiceBus;
     using Settings;
 
     public class FailOverNamespacePartitioning : INamespacePartitioningStrategy
     {
-        NamespaceConfigurations namespaces;
+        List<RuntimeNamespaceInfo> namespaces;
 
         public FailOverNamespacePartitioning(ReadOnlySettings settings)
         {
-            if (!settings.TryGet(WellKnownConfigurationKeys.Topology.Addressing.Namespaces, out namespaces))
-            {
-                throw new ConfigurationErrorsException($"The '{nameof(FailOverNamespacePartitioning)}' strategy requires exactly two namespaces to be configured, please use {nameof(AzureServiceBusTransportExtensions.NamespacePartitioning)}().{nameof(AzureServiceBusNamespacePartitioningSettings.AddNamespace)}() to register the namespaces.");
```

Comment:
We're losing the error message which would explain why strategy is not configured properly.